Documentation/bash-scripts.md

```diff
@@ -0,0 +1,52 @@
+# Bash scripts
+
+A common way to use acbuild is via a bash script. There's a common template
+that the examples for acbuild follow, and it makes cleanup a little easier. The
+template is at the end of this file, and here we explain the reasoning behind
+the parts of it.
+
+The 1st line, `#!/usr/bin/env bash`, is simply good form to make sure the
+script is executed with bash as opposed to the current shell.
+
+The 2nd line, `set -e` makes the script exit in the event of an error. If some
+command doesn't exit correctly, due to something like a typo in a command, or
+the command in `acbuild run` encountering an error, it is better to end the
+script, instead of continuing on and producing an ACI with unintended values or
+contents.
+
+The 4th through the 7th lines check if the script is being run as the root
+user. If it is not, the script prints an error and exits. This check may not be
+necessary, depending on the actions performed in the script.
+
+The 9th through 12th lines define a function that performs cleanup after the
```

Comment:
The rest of the script is concerned with cleanup and error handling. The `acbuildend` function is interesting, as it serves as a simple "catch" mechanism for the script, called on a `trap` triggered by either reaching the script's end, or  by a non-zero return value from any command in the script. `Acibuildend` stores the last command's exit code, then calls `acbuild end` to terminate the build, passing the exit code, if any, through the script exit, to help with troubleshooting.